kuksa_databroker/databroker/src/broker.rs

```diff
@@ -626,10 +627,14 @@ impl Subscriptions {
             }
         }
 
-        for sub in &self.change_subscriptions {
+        for sub in &mut self.change_subscriptions {
             match sub.notify(changed, db).await {
                 Ok(_) => {}
-                Err(err) => error = Some(err),
+                Err(err) => {
+                    error = Some(err);
+                    let taken_sender = take(&mut sub.sender);
+                    drop(taken_sender);
```

Comment:
I don't understand the logic of what is going on here at all... No need to use `mem::take` (to replace the sender with a "default" value?) only to immediately `drop` it.

Just set it to `None`, which I assume was your intention when changed it to `Option<Sender>`, right?

```suggestion
                    sub.sender = None:
```

Reply:
(or revert this part and drop subscriptions the same way it's done when a closed channel is detected, see general comment)

Reply:
your second suggestion makes the most sense from my POV. I will handle it in cleanup method if it gets called periodically and on error. Thanks for the remarks :)

Reply:
Awesome :wink: